### PR TITLE
Added Floating Action Button to scaffold layout

### PR DESCRIPTION
**Pull request by AnkitSuda**
Added Fab to scaffold layout so you don't have to wrap scaffold layout inside a box and put fab there.

How to add fab:

```
CollapsingToolbarScaffold(
    state = rememberCollapsingToolbarScaffoldState(), // provide the state of the scaffold
    toolbar = {
        // contents of toolbar go here...
    },
   floatingActionButton = {
       // contents of fab go here..
   },
   floatingActionButtonPosition = FabPosition.END, // position of fab
) {
    // main contents go here...
}
```

### CODE REVIEW

**Comment by onebone** [comment]
Thank you for having interest on the project! I like the idea of providing some popular components in the library. However I want to keep the *base* scaffold as minimal as possible and fab doesn't seem to be required to be implemented directly to the `CollapsingToolbarScaffold`. May I ask you to compose the fab implementation on top of the base scaffold, and create separate composable something like `ToolbarWithFabScaffold`?

**Comment by AnkitSuda** [comment]
Now FAB contents goes in ToolbarWithFabScaffold
CollapsingToolbarScaffold stays untouched

**Comment by AnkitSuda** [comment]
Now use CollapsingToolbarScaffold to add FAB

```
ToolbarWithFabScaffold(
    state = rememberCollapsingToolbarScaffoldState(), // provide the state of the scaffold
    toolbar = {
        // contents of toolbar go here...
    },
   fab = {
       // contents of fab go here..
   },
   fabPosition = FabPosition.END, // position of fab
) {
    // main contents go here...
}
```

File: lib/src/main/java/me/onebone/toolbar/ToolbarWithFabScaffold.kt [hunk 1]
```diff
@@ -0,0 +1,101 @@
+package me.onebone.toolbar
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.SubcomposeLayout
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun ToolbarWithFabScaffold(
```
Comment (onebone) [inline lib/src/main/java/me/onebone/toolbar/ToolbarWithFabScaffold.kt:10]: May I ask you to mark it as `@ExperimentalToolbarApi`? I'd like to put a few new components which may result in breaking function signature and behaviors.

File: lib/src/main/java/me/onebone/toolbar/FabPosition.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package me.onebone.toolbar
+
+
+@Suppress("INLINE_CLASS_DEPRECATED", "EXPERIMENTAL_FEATURE_WARNING")
+inline class FabPosition internal constructor(@Suppress("unused") private val value: Int) {
```
Comment (onebone) [inline lib/src/main/java/me/onebone/toolbar/FabPosition.kt:5]: What is the `value` property for? Is it inappropriate to replace it with enum?

**Review by onebone — CHANGES_REQUESTED** [review]

File: lib/src/main/java/me/onebone/toolbar/ToolbarWithFabScaffold.kt (hunk 1 above)
Comment (AnkitSuda) [reply lib/src/main/java/me/onebone/toolbar/ToolbarWithFabScaffold.kt:11]: Added in new commit

File: lib/src/main/java/me/onebone/toolbar/FabPosition.kt (hunk 2 above)
Comment (AnkitSuda) [reply lib/src/main/java/me/onebone/toolbar/FabPosition.kt:5]: That was in official material's scaffold component. I copied this class from there.
I can change it to Enum if you want.

[the remark this answers is not in the document]
Comment (AnkitSuda) [reply lib/src/main/java/me/onebone/toolbar/FabPosition.kt:5]: That was in official material's scaffold component. I copied this class from there.

Comment (onebone) [reply lib/src/main/java/me/onebone/toolbar/FabPosition.kt:5]: I'm purely curious why they made that decision. Possibly reducing the size of dex?
For now, though, I would like to change it to enum for the favor of enforcing exhaustive branches 😃

File: lib/src/main/java/me/onebone/toolbar/CollapsingToolbarScaffold.kt [hunk 3]
```diff
@@ -100,10 +100,10 @@ fun CollapsingToolbarScaffold(
 			minWidth = 0,
 			minHeight = 0,
 			maxHeight =
-				if(scrollStrategy == ScrollStrategy.ExitUntilCollapsed)
-					max(0, constraints.maxHeight - toolbarState.minHeight)
-				else
-					constraints.maxHeight
+			if(scrollStrategy == ScrollStrategy.ExitUntilCollapsed)
```
Comment (onebone) [inline lib/src/main/java/me/onebone/toolbar/CollapsingToolbarScaffold.kt:103]: May I ask you to revert this change?

**Review by onebone — CHANGES_REQUESTED** [review]

File: lib/src/main/java/me/onebone/toolbar/CollapsingToolbarScaffold.kt (hunk 3 above)
Comment (AnkitSuda) [reply lib/src/main/java/me/onebone/toolbar/CollapsingToolbarScaffold.kt]: Done

File: lib/src/main/java/me/onebone/toolbar/FabPosition.kt (hunk 2 above)
Comment (AnkitSuda) [reply lib/src/main/java/me/onebone/toolbar/FabPosition.kt]: Changed to Enum in latest commit

File: lib/src/main/java/me/onebone/toolbar/FabPosition.kt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package me.onebone.toolbar
+
+enum class FabPosition(val value: Int) {
```
Comment (onebone) [inline lib/src/main/java/me/onebone/toolbar/FabPosition.kt:3]: Lastly, can you remove `value` if not needed? Other things look good :+1:

**Review by onebone — CHANGES_REQUESTED** [review]

**Comment by onebone** [comment]
Great work 😃 Thank you for your contribution!
